libdnf5/conf/option_enum.cpp

```diff
@@ -26,165 +26,86 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 
 namespace libdnf5 {
 
-template <typename T>
-bool from_string(T & out, const std::string & in, std::ios_base & (*manipulator)(std::ios_base &)) {
-    std::istringstream iss(in);
-    return !(iss >> manipulator >> out).fail();
-}
-
-template <typename T>
-OptionEnum<T>::OptionEnum(ValueType default_value, const std::vector<ValueType> & enum_vals)
+class OptionEnum::Impl {
+public:
+    Impl(std::string default_value, const std::vector<std::string> & enum_vals)
+        : enum_vals(std::move(enum_vals)),
+          default_value(default_value),
```

Comment:
Replace `std::string default_value` by `const std::string & default_value`. Pass by reference. no need to copy.
It is copied aggain on the following lines.

Reply:
Thanks, updated.